### PR TITLE
Adding in 15 minute read-data for Surplus Generation

### PR DESCRIPTION
**Pull request by johnavich**
This helps to solve getting surplus generation data from the smartmetertexas website. This is to help generate data for the Energy Dashboard in HA, but i need to add this into the offical repo first.

### CODE REVIEW

File: smart_meter_texas/__init__.py [hunk 1]
```diff
@@ -35,9 +36,9 @@
     SmartMeterTexasRateLimitError,
 )
 
-__author__ = "Graham Wetzler"
-__email__ = "graham@wetzler.dev"
-__version__ = "0.4.7"
+__author__ = "Bobbie Turner"
```
Comment (grahamwetzler) [inline smart_meter_texas/__init__.py:39]: Please don't overwrite these. I will bump the version using a cli tool.

File: smart_meter_texas/__init__.py [hunk 2]
```diff
@@ -85,6 +87,55 @@ async def read_meter(self, client: Client):
...
+            energy = data["energyData"]
+        except KeyError:
+            _LOGGER.error("Error reading data: ", json_response)
+            raise SmartMeterTexasAPIError(
+                    f"Error parsing response: {json_response}"
+                )
+        else:
+            h=-1
+            minute=0
+            for E in energy:
```
Comment (grahamwetzler) [inline smart_meter_texas/__init__.py:119]: Please use variables names that meet the [PEP8 style](https://www.python.org/dev/peps/pep-0008/#function-and-variable-names).

**Review by grahamwetzler — CHANGES_REQUESTED** [review]
Please see requested changes. Also please run `black` against `__init__.py` before pushing.

File: smart_meter_texas/__init__.py (hunk 1 above)
Comment (johnavich) [reply smart_meter_texas/__init__.py:42]: Updated, sorry.

File: smart_meter_texas/__init__.py (hunk 2 above)
Comment (johnavich) [reply smart_meter_texas/__init__.py]: I believe i have resolved this, please let me know if i've missed something.

**Review by johnavich — COMMENTED** [review]
Requested changes have been made.

File: smart_meter_texas/__init__.py [hunk 3]
```diff
@@ -38,6 +39,7 @@
 __author__ = "Graham Wetzler"
 __email__ = "graham@wetzler.dev"
 __version__ = "0.4.7"
+__author__ = "Bobbie Turner"
```
Comment (grahamwetzler) [inline smart_meter_texas/__init__.py:42]:
```suggestion
```

**Review by grahamwetzler — CHANGES_REQUESTED** [review]
This file still needs `isort` and `black` run:

`isort . --profile black`
`black .`

File: smart_meter_texas/__init__.py (hunk 3 above)
Comment (johnavich) [reply smart_meter_texas/__init__.py]: Done

**Comment by johnavich** [comment]
Tests ran

**Review by grahamwetzler — APPROVED** [review]
Looks good. Thank you!

**Comment by goldbattle** [comment]
Thanks again for the project and PR to allow tracking of the surplus. Were you going to make a PR to the home assistant repo? It seems to be pinned at version `0.4.7`.
https://github.com/home-assistant/core/blob/dev/homeassistant/components/smart_meter_texas/manifest.json#L6

**Comment by goldbattle** [comment]
If you use utility meter in home assistant this would do the compounding for you.
https://www.home-assistant.io/integrations/utility_meter/

**Pull request merged**